jetty-server/src/main/java/org/eclipse/jetty/server/AbstractConnector.java

```diff
@@ -188,6 +189,9 @@ public AbstractConnector(
             pool = _server.getBean(ByteBufferPool.class);
         _byteBufferPool = pool != null ? pool : new ArrayByteBufferPool();
         addBean(_byteBufferPool);
+        RetainableByteBufferPool retainableByteBufferPool = _server.getBean(RetainableByteBufferPool.class);
+        retainableByteBufferPool = retainableByteBufferPool != null ? retainableByteBufferPool : new RetainableByteBufferPool();
+        addBean(retainableByteBufferPool);
```

Comment:
might be worthwhile to be explict here with the management attribute.  ie only if it is a new pool should we manage it.   I know this probably happens automagically, but no harm being explicit.

Reply:
I once again just copied the logic used for the `_byteBufferPool` field. Do we want a different one for `retainableByteBufferPool`?

Reply:
I think it is worthwhile being explicit, since the order of creating/starting pools might not be straight forward with backwards compatibility logic.

Reply:
Argh, can't do as `AbstractHTTP2ServerConnectionFactory` calls `getBean()` on the connector to get the `RetainableByteBufferPool` instance, so this bean must always be registered otherwise the legacy `ByteBufferPool` is going to be used instead.

Reply:
I'm not saying don't register it.  I'm saying register it with `addBean(Object, boolean)` so you can pass in the managed status of it.
Currently if you just add a bean that is itself a lifecycle, it gets added in automode.   Then when the container is started, if it also has to start the auto bean, it changes it to managed.  If the autobean is already started it is changed to not-managed.

I'm suggesting that if a pool is found on the server, you should `addBean(pool, false)` as we should never manage a server bean.  If we create a new pool, then it should be `addBean(pool, true)` so that we explicitly manage it.

Reply:
Something like:
```java
    RetainableByteBufferPool pool = _server.getBean(RetainableByteBufferPool.class);
    addBean(pool == null ? new RetainableByteBufferPool() : pool, pool == null);
```

```

Reply:
Oh okay, I didn't get that but it makes sense now. I'll make this change.